src/onegov/search/indexer.py

```diff
@@ -323,27 +323,41 @@ def __init__(self, mappings, queue, es_client, engine, session):
             'en': 'english',
         }
 
-    def index(self, task):
+    def index(self, tasks) -> bool:
         """ Update the 'fts_idx' column (full text search index) of the given
-        object.
+        object(s)/task(s).
+
+        In case of a bunch of tasks we are assuming they are all from the
+        same schema and table in order to optimize the indexing process.
+
+        :param tasks: A list of tasks to index
+        :return: True if the indexing was successful, False otherwise
         """
+        content = []
+
+        if not isinstance(tasks, list):
+            tasks = [tasks]
 
         try:
-            language = (
-                self.idx_language_mapping.get(task['language'], 'simple'))
-            data = {
-                k: str(v)
-                for k, v in task['properties'].items()
-                if not k.startswith('es_')}
-            schema = task['schema']
-            tablename = task['tablename']
-            id = task['id']
-            id_key = task['id_key']
+            for task in tasks:
+                language = (
+                    self.idx_language_mapping.get(task['language'], 'simple'))
+                data = {
+                    k: str(v)
+                    for k, v in task['properties'].items()
+                    if not k.startswith('es_')}
+                _id = str(task['id'])
```

Comment:
Have you tried leaving `task['id']` unmodified? I worry SQLAlchemy will do the wrong thing here when binding the parameter, because it doesn't know the type of the id column in the table. Technically the id should already have the correct Python type and be automatically converted to the correct database type.

It's possible automatic type coercion makes this a non-issue, but it's better to check with some models that use `int` and `UUID` primary keys, that this actually works (starting from an empty index column).

Reply:
Yes, unmodified works now